src/notation/view/pianokeyboard/pianokeyboardview.cpp

```diff
@@ -53,13 +55,25 @@ static QColor mixedColors(QColor background, QColor foreground, qreal opacity)
 }
 
 PianoKeyboardView::PianoKeyboardView(QQuickItem* parent)
-    : muse::uicomponents::QuickPaintedView(parent), m_controller(new PianoKeyboardController())
+    : muse::uicomponents::QuickPaintedView(parent), muse::Injectable(muse::iocCtxForQmlObject(this))
+{
+    setAcceptedMouseButtons(Qt::LeftButton);
+}
+
+PianoKeyboardView::~PianoKeyboardView()
+{
+    delete m_controller;
```

Comment:
I'm not sure but perhaps we should make this a bit safer, just in case the destructor is already called before `init()` has ever been called:
```suggestion
    if (m_controller) {
        delete m_controller;
    }
```

Reply:
there is no problem calling delete on a null pointer, and if it is not null, then if will pass it